### ISSUE TITLE
`pihole -q` does not consider regex.list

### ISSUE THREAD

**Issue by 0xbs** [pi-hole/pi-hole#2539]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**
1

---
**Expected behaviour:**

`pihole -q www.facebook.com` should show the domain as blocked if it matches a regex in `/etc/pihole/regex.list` such as `(^|\.)(facebook|fb|fbcdn|fbsbx|tfbnw)\.(com|net)$`

**Actual behaviour:**

    $ pihole -q www.facebook.com
    [i] No results found for www.facebook.com within the block lists

**Steps to reproduce:**

```
$ cat /etc/pihole/regex.list
(^|\.)(facebook|fb|fbcdn|fbsbx|tfbnw)\.(com|net)$
$ pihole -q www.facebook.com
[i] No results found for www.facebook.com within the block lists
```

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**
Will only provide if cannot be reproduced.
```
*** [ DIAGNOSING ]: Core version
[i] Core: v4.0 (https://discourse.pi-hole.net/t/how-do-i-update-pi-hole/249)
[i] Branch: master
[i] Commit: v4.0-0-gddbdb51

*** [ DIAGNOSING ]: Web version
[i] Web: v4.0 (https://discourse.pi-hole.net/t/how-do-i-update-pi-hole/249)
[i] Branch: master
[i] Commit: v4.0-0-gaf8c926

*** [ DIAGNOSING ]: FTL version
[✓] FTL: v4.0

*** [ DIAGNOSING ]: dnsmasq version
[i] 340:

*** [ DIAGNOSING ]: lighttpd version
[i] 1.4.45

*** [ DIAGNOSING ]: Pi-hole processes
[✗] dnsmasq daemon is inactive
[✓] lighttpd daemon is active
[✓] pihole-FTL daemon is active
```

**Troubleshooting undertaken, and/or other relevant information:**
n/a

**Comment by Mcat12** [comment]
This PR attempts to solve this, but has run into issues and is stalled: #2346

**Comment by no-replies** [comment]
(.*?)(facebook|fb|fbcdn|fbsbx|tfbnw)\.(com|net|me)$

**Comment by mmotti** [comment]
@Mcat12  / @dschaper  this one can be marked down as **fixed in next release**

**Comment by Mcat12** [comment]
Fixed by #2780

**Comment by MeisterLuchs** [comment]
Just to be clear: Is this the same problem that is causing Pi-hole to not block domains that should be blocked by a RegEx-string specified in the `regex.list` or should I open a new issue?

**Comment by DL6ER** [comment]
This issue is only about `pihole -q` not showing domains as being blocked even though there are matching regex filter that **are** used for blocking by Pi-hole.

Note that we rewrote the entire domain handling from list(file)-based to table(database)-based.
Go ahead and open a new issue so we can clarify what is the issue you're seeing and whether this is already fixed in the upcoming v5.0.

**Comment by DL6ER** [comment]
Pi-hole v5.0 has just been released.
